Payjoin

yashrajd opened the pull request.

yashrajd commented:
> Just a heads-up, for the screens, you can use the image gallery to show the full-sized mock-ups in a horizontal slider, like [here](https://deploy-preview-993--bitcoin-design-site.netlify.app/guide/daily-spending-wallet/contacts/#adding-a-contact).

It does look great, Stephen suggested it as well. Will do.

sbddesign commented:
@yashrajd and I have been jamming on header images. We have some rough draft ideas. Looking for some opinions on these ideas, then we can actually clean up one of these and make it work as a header. Or suggest ideas if you have them.

### Concept: cyberpunks making a secret transaction in a dark alley
![header-cyberpunk-transaction](https://user-images.githubusercontent.com/2414177/224361137-ceaf7f56-cf82-41d4-a6d1-62926413adcc.jpg)

### Concept: a handshake between two characters (payjoins are a collaborative process)
![header-handshake](https://user-images.githubusercontent.com/2414177/224361139-9545b5e1-878a-4925-8aa3-66e49e1a5192.jpg)

### Concept: try to make some visual graphical metaphors for privacy, how payjoins obfuscate things, etc
![header-graphic-metaphor](https://user-images.githubusercontent.com/2414177/224361142-2191d3a6-75ce-4d2a-a6c1-8e8aa6a56f51.jpg)

### Concept: show Bob shopping for Alice's wedding ring
![header-bob-buying-ring](https://user-images.githubusercontent.com/2414177/224361144-c171c425-7ed8-4b2b-8c7a-349015cb2b7c.jpg)

DanGould commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,317 @@
+
+## Introduction
+
+Payjoin transactions can be a powerful tool providing privacy, fee-savings and UTXO consolidation benefits to users. They also present challenges and trade-offs due to their synchronous, multi-party nature.
```
```suggestion
Payjoin transactions can be a powerful tool providing privacy, fee-savings and UTXO consolidation benefits to users. They also present challenges and trade-offs due to their interactive nature.
```

I think all but self-spend transactions  are multi-party. They key with payjoin is that  it is *interactive*. Synchronicity may be considered a detail.

DanGould commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,317 @@
+
+It is also called pay-to-endpoint (P2EP) since the coordination is mediated through an endpoint run by the receiver. The endpoint address is communicated through a BIP-21 URI along with the payment address and amount.
+
+In the payjoin process, parties edit, sign and pass iterations of the transaction between each other before a final version is broadcasted. [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Respecting_the_minimum_relay_fee_policy) codifies a protocol with 2 iterations, as shown in the visual here.
```
```suggestion
In the payjoin process, parties edit, sign and pass iterations of the transaction between each other before a final version is broadcasted. [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Respecting_the_minimum_relay_fee_policy) codifies a protocol with one round of interaction beyond address sharing, as shown in the visual here.
```

In studying interactive protocols, I've always seen and have come to expect interaction quantified in "rounds" from round-robin tournament terminology.

yashrajd replied:
Someone else (or you) made the point in the google doc few days back. In an extended payjoin scenario where the sender edits/signs and sends it back to receiver to sign/broadcast the tx, will it be considered 1.5 rounds?

yashrajd replied:
I'm open to adding this as an alternative phrasing...

DanGould commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,317 @@
+View this flow in [Figjam here](https://www.figma.com/file/69uUDWVc8N9t5Bej8pZEsF/PayJoin-User-Flows?node-id=0%3A1&t=8eOBQa7JofjZ3p3z-1). The detailed process and considerations can be reviewed in this [FigJam file](https://www.figma.com/file/NzMvwyzP7x5jfGmwNUKRov/PayJoin-Process-Flows?node-id=336%3A619&t=BhGymSWCzl2Xm2oO-1).
+
+
+##### The Payjoin handshake
+
+<p class="h3">The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.</p>
+
+The above steps are not important to the users involved in the transaction, and do not need specific user action from either the sender or receiver. More generally, the transacting parties need not even be aware that this is occurring.
```
I  love to see innovation in the way we can explain the interaction! Way easier to understand what's going on if we can  put consise words to it. However, I think "payjoin *rendezvous*" is better suited than "payjoin handshake" in this context.

In the context of an interactive protocol (like TLS), a handshake refers to a series of steps or messages exchanged between two parties to establish a connection and initialize communication. In payjoin, the communication is already over after the steps described here as a handshake are complete

A rendezvous refers to a mechanism or procedure for two parties to synchronize and agree on the parameters of an interaction.

yashrajd replied:
I can see the confusion for a technical reader but the 'handshake' is intended for a colloquial, non-technical purpose/audience. It's also clearer & simpler (to me) than the 'rendezvous'.

DanGould replied:
I'm not even sold on rendezvous. It would be good to say *why* the term needs to be introduced. Did you run into confusion in existing UI where there was no explanation of what was going on? And introducing new vocabulary creates a better experience?

yashrajd replied:
I think so, addressed this in the case study text.

DanGould commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,317 @@
+
+#### Investigation: BTCPay merchant/receiving
+
+The author successfully tested receiving payjoins (BTCPay 1.7.12) in Feb 2023.
```
```suggestion
The author successfully tested receiving payjoins (BTCPay v1.7.12) in Feb 2023.
```

DanGould reviewed (changes requested):
Love to see this in draft. If I were a maintainer, I'd like to see this merged after these changes as long as there was some understanding that it would continue to be revised. 

I left a few suggestions, mostly to describe the protocol as interactive in contrast to typical non-interactive bitcoin flow.

Great 🚀

yashrajd commented:
<img width="1090" alt="image gallery" src="https://user-images.githubusercontent.com/114083781/224522929-6b72d95b-147a-4036-9d9b-5326e28bc819.png">

GBKS commented:
@sbddesign @yashrajd great explorations for the header images. I'd recommend staying away from anything that implies secrecy, fear, having things to hide or even illegal activity (dark alley, guy in trench coat checking to see if he's being followed). Privacy should be a natural default for everyone. So I Iike the handshake one better. I also like the idea of the third one, where someone tries to snoop and just can't figure things out. Maybe those could be mashed up? Two people shaking hands with some visual of a transaction arising from that, and then there being tons of transactions around it that look identical. If it's too much to pack into an image, maybe consider that the page title and image are typically seen together each each one could communicate one aspect of what you want to express.

GBKS commented:
@DanGould noted your point about merging with the expectation of revisions, it's a good way to go about things. @yashrajd, this is totally up to you. The PR status is draft, and your top comment states it is not ready, so I have not done a detailed review yet. Feel free to update that whenever works for you.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+For the purpose of creating the sender flow, we will assume that the receiver only contributes UTXOs but does not contribute to the fees since this allows a simple, automated process for both parties.
+
+### In short, the sender flow outlined here asks the user to choose a fee-range instead of a fee-amount (or fee-rate) while keeping the rest of the user flow almost exactly the same. We use it to set 3 optional parameters specified in BIP-78, which can be used to construct a simple but effective payjoin implementation.
```
```suggestion
** In short, the sender flow outlined here asks the user to choose a fee-range instead of a fee-amount (or fee-rate) while keeping the rest of the user flow almost exactly the same. We use it to set 3 optional parameters specified in BIP-78, which can be used to construct a simple but effective payjoin implementation.**
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+##### The Payjoin handshake
+
+### The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.</p>
```
```suggestion
** The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.**
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+##### Prototypes for the Payjoin sender flow
+
+### The intention is to keep the user-flow as close to the default flow as possible. Where it becomes necessary to deviate, we educate and help the user to navigate the UI.
```
```suggestion
**The intention is to keep the user-flow as close to the default flow as possible. Where it becomes necessary to deviate, we educate and help the user to navigate the UI.**
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+### The intention is to keep the user-flow as close to the default flow as possible. Where it becomes necessary to deviate, we educate and help the user to navigate the UI.
+
+<div class="image-slide-gallery">
```
Don't need this line, the `image-gallery.html` include already has it.

yashrajd replied:
```suggestion

```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+{% include image-gallery.html pages = page.sender-flow-screens %}
+
+</div>
```
Don't need this line, the `image-gallery.html` include already has it.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+### Receiving Payjoins
+
+Based on the BIP-78 protocol, [the receiver has higher requirements](http://0.0.0.0:4000/guide/case-studies/payjoin-case-study/#understanding-users--requirements) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions on other hand could implement a payjoin receiver setup easily, but avoid hot wallets due to secu ...
```
```suggestion
Based on the BIP-78 protocol, [the receiver has higher requirements]({{ '/guide/case-studies/payjoin-case-study/#understanding-users--requirements' | relative_url }}) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions on other hand could implement a payjoin receiver setup easily, but avoid hot wallets due to security concerns.
```
Don't want to link to your local dev environment here.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+Based on the BIP-78 protocol, [the receiver has higher requirements](http://0.0.0.0:4000/guide/case-studies/payjoin-case-study/#understanding-users--requirements) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions on other hand could implement a payjoin receiver setup easily, but avoid hot wallets due to secu ...
+
+### While practically every mobile wallet is a hot wallet and it is trivial to fund it, having an online server where the payjoin handshake can be performed is difficult for technical and practical reasons. This might be the biggest impediment to payjoin support and adoption so far.
```
```suggestion
**While practically every mobile wallet is a hot wallet and it is trivial to fund it, having an online server where the payjoin handshake can be performed is difficult for technical and practical reasons. This might be the biggest impediment to payjoin support and adoption so far.**
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+Here are problems identified:
+
+- Generating a payjoin-enabled payment link is automatic once it is setup
```
Why is it a problem that it's automatic?

yashrajd replied:
Fixed.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+#### Proposed Receiver Flow
+
+Here we will devise user flows for a POS system that can always be online. This excludes mobile wallets.
```
```suggestion
Here we will devise user flows for a point-of-sale (POS) system that can always be online. This excludes mobile wallets.
```
Make sure to explain abbreviations when you introduce them.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+A receiver should be able to set up payjoin during onboarding on the platform (app or POS system) or any time after that. The following is a standalone payjoin setup flow.
+
+{% include picture.html
+   image = "assets/images/guide/case-studies/payjoin-case-study/payjoin-receiver-flow@2x.png"
```
```suggestion
   image = "assets/images/guide/case-studies/payjoin-case-study/payjoin-receiver-flow.png"
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+- Payjoin setup workflow
+- Additional payjoin features like third party endpoints, address substitution, cold-wallet backfill
+- Monitor payjoin readiness: hot wallet status, pj-endpoint status etc.
```
```suggestion
- Monitor payjoin readiness: hot wallet status, Payjoin-endpoint status etc.
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+We learnt many things over the course of this case study, and the user flows we have designed already use many of these insights.
+
+- Payjoins are a great way to consolidate UTXOs so they pay lower fees in the long run – this can be especially useful to exchanges, trading desks who need this the most and possess technical ability to implement a robust setup
```
```suggestion
- Payjoins are a great way to consolidate UTXOs so participants pay lower fees in the long run. This can be especially useful for trading desks at exchanges who need this the most, and possess the technical ability to implement a robust setup
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+We learnt many things over the course of this case study, and the user flows we have designed already use many of these insights.
+
+- Payjoins are a great way to consolidate UTXOs so they pay lower fees in the long run – this can be especially useful to exchanges, trading desks who need this the most and possess technical ability to implement a robust setup
+- Due to the back-&-forth between parties which can extend over multiple iterations/rounds, this case study introduces the term ‘payjoin handshake’ to encompass all the events between user actions
```
```suggestion
- Due to the back-and-forth between parties which can extend over multiple iterations/rounds, this case study introduces the term ‘Payjoin handshake’ to encompass all the events between user actions
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+- Payjoins are a great way to consolidate UTXOs so they pay lower fees in the long run – this can be especially useful to exchanges, trading desks who need this the most and possess technical ability to implement a robust setup
+- Due to the back-&-forth between parties which can extend over multiple iterations/rounds, this case study introduces the term ‘payjoin handshake’ to encompass all the events between user actions
+- Maintaining an always-online endpoint seems to be the biggest hurdle for payjoin implementation, we list some alternatives here
+- Payjoin burdens the receiver with more requirements but also provides more benefits and opportunities
```
```suggestion
- Payjoin burdens the receiver with more requirements, but also provides more benefits and opportunities
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+- Maintaining an always-online endpoint seems to be the biggest hurdle for payjoin implementation, we list some alternatives here
+- Payjoin burdens the receiver with more requirements but also provides more benefits and opportunities
+- Address substitution is a powerful idea from BIP-78: it is a risk that can become a powerful ally for the payjoiner
+- Payjoins receivers can leverage their daily spending wallet-saving wallet setup, where the daily spending is used by the receiver as the payjoin hot wallet
```
```suggestion
- Instead of setting up another wallet, Payjoin receivers can leverage their daily spending wallet, which is used by the receiver as the Payjoin hot wallet
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+- Payjoin burdens the receiver with more requirements but also provides more benefits and opportunities
+- Address substitution is a powerful idea from BIP-78: it is a risk that can become a powerful ally for the payjoiner
+- Payjoins receivers can leverage their daily spending wallet-saving wallet setup, where the daily spending is used by the receiver as the payjoin hot wallet
+- Payjoin implementations should be mindful of the round-fee-rate heuristic for identifying payjoins, handle it without need for user decisions
```
```suggestion
- Payjoin implementations should be mindful of the round-fee-rate heuristic for identifying Payjoins, and handle it without a need for complex user decisions
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+## New Horizons
+
+Payjoins can be complicated to understand or implement given the multiple parties and moving parts (such as the endpoint) involved, but this also opens the door for multiple features and services.
```
```suggestion
Payjoins can be complicated to understand or implement, given the multiple parties and moving parts (such as the endpoint) involved. But, this also opens the door for multiple features and services.
```

yashrajd replied:
Agree it is kinda long as a bullet point, but I avoid using periods in bullet points.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+Here are some examples:
+
+- Third party endpoints could be used by receivers (turning-off address substitution), trading some privacy for convenience
```
```suggestion
- Third-party endpoints could be used by receivers (turning off address substitution), trading some privacy for convenience
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+Here are some examples:
+
+- Third party endpoints could be used by receivers (turning-off address substitution), trading some privacy for convenience
+- There is exciting work underway around [serverless](https://gist.github.com/DanGould/243e418752fff760c9f6b23bba8a32f9) payjoin implementation that would work great on platforms with both bitcoin & lightning
```
```suggestion
- There is exciting work under way around [serverless](https://gist.github.com/DanGould/243e418752fff760c9f6b23bba8a32f9) Payjoin implementations that would work great on platforms implementing both bitcoin & lightning
```

yashrajd replied:
@sbddesign and I settled on using payjoin like we use bitcoin, will incorporate the other suggestion.

GBKS replied:
Great. Still seeing lots of capitalized used of payjoin in sentences throughout the page, so I'll wait with reviewing.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+- There is exciting work underway around [serverless](https://gist.github.com/DanGould/243e418752fff760c9f6b23bba8a32f9) payjoin implementation that would work great on platforms with both bitcoin & lightning
+- Receiver-side implementations could use address substitution to let users make payments of their own
+- Payjoin coordination over NFC (or other wireless near-range communication technology)
+- BIP-78 could to allow 1 more iteration where sender can adjust fees post-handshake & send it back to receiver to broadcast
```
```suggestion
- BIP-78 could to allow 1 more iteration where senders can adjust fees post-handshake & send the transaction back to receivers to broadcast
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+description: <>
+nav_order: 5
+parent: Case studies
+permalink: /guide/case-studies/payjoin-case-study/
```
Could you please move this page to `/guide/case-studies/payjoin/`? No need to repeat `case study` twice in the URL. That also means moving files and folders around and updating various URLs in this PR.

mouxdesign commented:
Really like the headings you guys came up with! 

Some minor suggestions to text from my side. Other than that, looks good! Nicely done!

## Introduction:
Text: However, real time coordination between the sender & receiver mediated through an endpoint is required to construct them.
Suggestion: Real time coordination between the sender & receiver mediated through an endpoint is required to construct them.

Text: Proposes specific user flows and where possible, working designs for the same.
Suggestion: Proposes specific user flows and where possible as well as design solutions.

## Our Approach
Text: This case study follows these tentative steps:
Suggestion: This case study follows these steps:

Text: sender and receiver of a payment
Suggestion: Sender and receiver of a payment

## Proposed Receiver Flow 
Text: Here we will devise user flows for a POS system that can always be online. This excludes mobile wallets.
Suggestion: We will created a  user flows for a POS system that can always be online. This excludes mobile wallets.

## New Horizons
Text: This case study not only tackles the design aspects around Payjoin, it hopes to boost interest in the various use cases and benefits.
Suggestion: This case study tackles the design aspects around Payjoin and it hopes to boost interest in it's various use cases and benefits.

DanGould commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+Here are the findings:
+
+- The fee rate drops once transaction size increases during the payjoin process,  which may change estimated confirmation time
+- The flow results in a transaction that has a round amount fee-rate after removing one input, which may be used by chain analysis firms to spot payjoins
```
See "Round payment amount" used to distinguish between change / payment [here](https://blockstream.info/tx/b2ddf68bf2318b51ec5e6bdc7831e7063cd04c0ff84edbb56808de62fbb0de74)

I believe the case study hightlights how a similar round fee-rate heuristic may distinguish the payment amount from change here.

https://en.bitcoin.it/wiki/Privacy#Round_numbers

yashrajd replied:
Thank you, do you suggest any changes/clarifications to my wording?

yashrajd commented:
> Really like the headings you guys came up with!
> 
> Some minor suggestions to text from my side. Other than that, looks good! Nicely done!

Thank you, I appreciate the time and detailed reading put into this!
> 
> ## Introduction:
> Text: However, real time coordination between the sender & receiver mediated through an endpoint is required to construct them. Suggestion: Real time coordination between the sender & receiver mediated through an endpoint is required to construct them.
> 
Both the requirements, namely "real time coordination" and "an endpoint" places constraints on the UX, which is why I used "however".

> Text: Proposes specific user flows and where possible, working designs for the same. Suggestion: Proposes specific user flows and where possible as well as design solutions.
> 
Not sure what you mean...
> ## Our Approach
> Text: This case study follows these tentative steps: Suggestion: This case study follows these steps:
> 
Makes sense. I think Christoph pointed this out too, so believe I have addressed it.
> Text: sender and receiver of a payment Suggestion: Sender and receiver of a payment
> 
> ## Proposed Receiver Flow
> Text: Here we will devise user flows for a POS system that can always be online. This excludes mobile wallets. Suggestion: We will created a user flows for a POS system that can always be online. This excludes mobile wallets.
> 
I used 'devise' to indicate thought and intention went into it and not simply created from vacuum.
 
> ## New Horizons
> Text: This case study not only tackles the design aspects around Payjoin, it hopes to boost interest in the various use cases and benefits. Suggestion: This case study tackles the design aspects around Payjoin and it hopes to boost interest in it's various use cases and benefits.
This is definitely an improvement. Incorporated!

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+redirect_from:
+ - /guide/payments/contacts/
+main_classes: -no-top-padding
+image: https://bitcoin.design/assets/images/guide/case-studies/blixt-wallet/blixt-wallet-preview.jpg
```
Need help fixing this.

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+<!-----
+Conversion notes:
+
+* Docs to Markdown version 1.0β34
```
Fixed.

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+
+---
+
+## Introduction
```
Great suggestion.

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,352 @@
+   height = 939
+%}
+
+<div markdown="1">
```
Not sure how to address this.

yashrajd commented:
Ready to merge if all looks good.

mouxdesign commented on guide/case-studies/payjoin.md:
```diff
@@ -109,7 +109,7 @@ In the payjoin process, parties edit, sign and pass iterations of the transactio
-The payjoin case study has its origins in a [design challenge](https://bitcoin.design/guide/resources/design-challenges/#challenge-6-private-purchase), which entailed creating a sender flow for payment using a Payjoin transaction.
+The payjoin case study has its origins in a [design challenge]({{ '/guide/resources/design-challenges/#challenge-6-private-purchase' | relative_url }}), which entailed creating a sender flow for payment using a Payjoin transaction.
 
 While payjoin provides privacy improvements and potential fee-savings, they present the challenge of coordination between transacting parties and may require new user flows to implement features enabled through [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki).
```
```suggestion
While payjoin provides privacy improvements and potential fee-savings, they present  unique challenges. One of these is the need of coordination between transacting parties to implement features through [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki). This would require modifications to the userflow. 
```

mouxdesign commented on guide/case-studies/payjoin.md:
```diff
@@ -240,7 +240,7 @@ View this flow in [Figjam here](https://www.figma.com/file/69uUDWVc8N9t5Bej8pZEs
 ##### The Payjoin handshake
 
-### The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.</p>
+** The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.**
```
```suggestion
** The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and sign the new draft of the transaction.**
```

mouxdesign commented on guide/case-studies/payjoin.md:
```diff
@@ -240,7 +240,7 @@ View this flow in [Figjam here](https://www.figma.com/file/69uUDWVc8N9t5Bej8pZEs
-### The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.</p>
+** The term ‘Payjoin handshake’ is being introduced to succinctly describe the process of sending a original PSBT to receiver’s endpoint, receiver contributing their inputs, modifying the transaction as needed and signing the new draft of the transaction.**
 
 The above steps are not important to the users involved in the transaction, and do not need specific user action from either the sender or receiver. More generally, the transacting parties need not even be aware that this is occurring.
```
```suggestion
The above steps are not important to the users involved in the transaction, and do not need specific user action from either the sender or receiver. More generally, the transacting parties do not even be aware that this is occurring.
```

mouxdesign commented on guide/case-studies/payjoin.md:
```diff
@@ -261,9 +260,9 @@ A detailed explainer of each screen can be viewed [here](https://docs.google.com
 ### Receiving Payjoins
 
-Based on the BIP-78 protocol, [the receiver has higher requirements](http://0.0.0.0:4000/guide/case-studies/payjoin-case-study/#understanding-users--requirements) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions on other hand could implement a payjoin receiver setup easily, but avoid hot wallets due to secu ...
+Based on the BIP-78 protocol, [the receiver has higher requirements]({{ '/guide/case-studies/payjoin-case-study/#understanding-users--requirements' | relative_url }}) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions on other hand could implement a payjoin receiver setup easily, but avoid hot wallets due to  ...
```
```suggestion
Based on the BIP-78 protocol, [the receiver has higher requirements]({{ '/guide/case-studies/payjoin-case-study/#understanding-users--requirements' | relative_url }}) than the sender does. While senders need a compatible mobile wallet, they cannot serve an always-online endpoint. Businesses & institutions could implement a payjoin receiver setup easily, but avoid hot wallets due to security concerns.
```

mouxdesign reviewed (commented):
Added in some text suggestions.

yashrajd reviewed (commented):
Responses & proposed revisions to @mouxdesign 's suggestions.

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+parent: Case studies
+permalink: /guide/case-studies/payjoin/
+main_classes: -no-top-padding
+image: https://bitcoin.design/assets/images/guide/case-studies/blixt-wallet/blixt-wallet-preview.jpg
```
```suggestion
image: "assets/images/guide/case-studies/payjoin-case-study/header-graphic-metaphor-preview-image.png"
```

GBKS replied:
You probably don't want the `"` at the start and end of the file name

GBKS commented on _compress_images_cache.yml:
```diff
@@ -2148,3 +2148,179 @@ assets/images/social-media/social-media-mobile.jpg: social-media-mobile.jpg
 assets/images/social-media/social-media-mobile@2x.jpg: social-media-mobile@2x.jpg
 assets/images/social-media/social-media-preview.jpg: social-media-preview.jpg
 assets/images/profile-picture.jpg: profile-picture.jpg
+assets/images/guide/case-studies/payjoin case study/BIP-78 process.png: BIP-78 process.png
```
Could you please remove the outdated cache lines here? Looks like you changed the folder name twice. The cache script is not clever enough to get rid of lines for images that were moved or removed.

yashrajd replied:
done.

GBKS commented on assets/images/guide/case-studies/payjoin/sender-flow-prototype.png:
Don't see this file used in the page. Make sure to only submit images that are being used to keep the repo tidy.

yashrajd replied:
I did remove several unused images but missed this one. Thank you!

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+This case study:
+
+- Outlines the incentives and objectives for both senders & receivers
+- Analyses the Payjoin UX on both sides and proposes approaches to design Payjoin experiences.
```
```suggestion
- Analyses the payjoin UX on both sides and proposes approaches to design payjoin experiences.
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+The sender on the other hand needs little-to-no setup (apart from using a supporting wallet) and can achieve an identical or near-identical user experience to a regular transaction, depending on their choices and the sophistication of the implementation.
+
+
+### What is a Payjoin?
```
```suggestion
### What is a payjoin?
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+### What is a Payjoin?
+
+In a payjoin, both the sender and receiver contribute inputs to the transaction in a coordinated manner. The payjoin mechanism is also called pay-to-endpoint (P2EP) since the coordination is mediated through an endpoint run by the receiver. The endpoint address is communicated through a [BIP-21](https://bitcoinqr.dev) URI along with the payment address and amount.
```
```suggestion
In a payjoin, both the sender and receiver contribute inputs to the transaction in a coordinated manner. The payjoin mechanism is also called pay-to-endpoint (P2EP), since the coordination is mediated through an endpoint run by the receiver. The endpoint address is communicated through a [BIP-21](https://bitcoinqr.dev) URI, along with the payment address and amount.
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+In a payjoin, both the sender and receiver contribute inputs to the transaction in a coordinated manner. The payjoin mechanism is also called pay-to-endpoint (P2EP) since the coordination is mediated through an endpoint run by the receiver. The endpoint address is communicated through a [BIP-21](https://bitcoinqr.dev) URI along with the payment address and amount.
+
+In the payjoin process, parties edit, sign and pass iterations of the transaction between each other before a final version is broadcasted. [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Respecting_the_minimum_relay_fee_policy) codifies a protocol with 2 iterations (or one round of interaction beyond address sharing), as shown in the visual here.
```
```suggestion
In the payjoin process, parties edit, sign and pass iterations of the transaction between each other, before a final version is broadcasted. [BIP-78](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Respecting_the_minimum_relay_fee_policy) codifies a protocol with 2 iterations (or one round of interaction beyond address sharing), as shown in the visual here.
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+This case study followed these steps:
+
+- Understanding users & technical requirements for senders & receivers
+- Exploring the payjoin implementations: BlueWallet (sending) and BTCPay Server (receiving)
```
```suggestion
- Exploring the payjoin implementations: [BlueWallet](https://bluewallet.io) (sending) and [BTCPay Server](https://btcpayserver.org) (receiving)
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+### Understanding users & requirements
+
+Technically, everyone conducting transactions on the Bitcoin blockchain can use payjoin. This case study categorises users in the following ways:
```
```suggestion
Technically, everyone conducting transactions on the bitcoin blockchain can use payjoin. This case study categorises users in the following ways:
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+Technically, everyone conducting transactions on the Bitcoin blockchain can use payjoin. This case study categorises users in the following ways:
+
+1. Institutions (businesses, exchanges etc.) and individuals
+2. sender and receiver of a payment
```
```suggestion
2. Sender and receiver of a payment
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+</div>
+
+
+### Sending Payjoins
```
```suggestion
### Sending payjoins
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+- The fee rate drops once transaction size increases during the payjoin process, which may change estimated confirmation time
+- The flow results in a transaction that has a round fee-rate after removing one input, which may be used by chain analysis firms to spot payjoins
+
+#### The Payjoin Sender Flow
```
```suggestion
#### The payjoin sender flow
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+The above steps are not important to the users involved in the transaction, and do not require specific user action from either the sender or receiver. More generally, the transacting parties need not even be aware that this is occurring, but must ensure they remain online during the process.
+
+
+##### Prototypes for the Payjoin sender flow
```
```suggestion
##### Prototypes for the payjoin sender flow
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+##### Prototypes for the Payjoin sender flow
+
+**The intention is to keep the user-flow as close to the default flow as possible. Where it becomes necessary to deviate, we educate and help the user to navigate the UI.**
```
```suggestion
**The intention is to keep the user flow as close to the default flow as possible. Where it becomes necessary to deviate, we educate and help the user to navigate the UI.**
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+
+
+### Receiving Payjoins
```
```suggestion
### Receiving payjoins
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+Here we will devise user flows for a point-of-sale (POS) system that can always be online. This excludes mobile wallets.
+
+A receiver should be able to set up payjoin during onboarding on the platform (app or POS system) or any time after that. The following is a standalone payjoin setup flow.
```
```suggestion
A receiver should be able to set up payjoins during onboarding on the platform (app or POS system), or any time after that. The following is a standalone payjoin setup flow.
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+- Payjoin implementations should be mindful of the round-fee-rate heuristic for identifying payjoins, handle it without need for user decisions
+
+
+## New Horizons
```
```suggestion
## New horizons
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+- There is exciting work underway around [serverless](https://gist.github.com/DanGould/243e418752fff760c9f6b23bba8a32f9) payjoin implementation that would work great on platforms implementing both bitcoin & lightning
+- Receiver-side implementations could use address substitution to let users make payments of their own
+- Payjoin coordination over NFC (or other wireless near-range communication technology)
+- BIP-78 could to allow 1 more iteration where the sender can adjust fees post-handshake & send it back to the receiver to broadcast
```
```suggestion
- BIP-78 could be used to allow 1 more iteration where the sender can adjust fees post-handshake & send it back to the receiver to broadcast
```

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+This case study tackles the design aspects around payjoin and it hopes to boost interest in it's various use cases and benefits.
+
+*Gratitude:*
+- *[Pavlenex](https://pavlenex.com/) from BTCPay for providing access to a Payjoin-capable btcpay instance to test receiving*
```
```suggestion
- *[Pavlenex](https://pavlenex.com/) from BTCPay for providing access to a payjoin-capable BTCPay instance to test receiving*
```

GBKS reviewed (changes requested):
Looks a ton tighter with this last iteration. I left a bunch more comments, mostly around lowercasing payjoin (as you and Stephen decided on), commas, and stuff like that.

yashrajd commented on guide/case-studies/introduction.md:
```diff
@@ -26,6 +26,11 @@ This was a collaboration between Blixt wallet and the Bitcoin Design Community t
 
 ---
 
+### [The Payjoin Experience]({{ '/guide/case-studies/payjoin' | relative_url }})
```
I get the choice to use sentence case in titles for the guide in general, but request case studies be exempt from this policy. Case studies are largely standalone works in my opinion and their titles are likely to be chosen as if they were an independent paper. I certainly did. Request you to reconsider.

GBKS replied:
Seems like a bit of a minor and random thing to stray from the standard format, considering everything else adheres, but sure thing. Just make sure the use is consistent throughout the whole page.

yashrajd commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,333 @@
+
+A bitcoin transaction where the sender and receiver collaboratively construct a transaction is called a Payjoin. Payjoin transactions can be a powerful tool providing privacy, fee-savings and UTXO consolidation benefits to users. They also present challenges and trade-offs due to their interactive & synchronous nature.
+
+**Payjoin transactions break some of the chain analysis [heuristics](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#motivation) during the course of making a payment. However, real time coordination between the sender & receiver mediated through an endpoint is required to construct them.**
```
"Real time" is valid usage. Accepting suggestion which is valid as well.

yashrajd commented on guide/case-studies/blixt-wallet.md:
```diff
@@ -165,11 +165,11 @@ We had a grand time re-designing this wallet interface. We learned a lot about t
    previousName = "Case studies"
-   nextUrl = "/guide/resources/"
-   nextName = "Resources"
+   nextUrl = "/guide/case-studies/payjoin/"
```
This block is the only change I made to this file. Not sure why some other lines show changes as well.

yashrajd replied:
This block is the only place I changed this file personally.

yashrajd commented on guide/case-studies/blixt-wallet.md:
```diff
@@ -32,11 +32,11 @@ Illustration sources: https://www.figma.com/file/eyQDCaMoQvnRVcHzSsgrzs/Blixt-UX
 ## How do we help users new to lightning start using their wallet?
 
-The [Bitcoin Design Community]({{ site.slack_invite_url }}) ran a design sprint on an open-source lightning wallet.  During this sprint, we collaboratively focused on re-designing one aspect of the user flow, the **onboarding journey**. In this collaborative effort, [we got together](https://www.youtube.com/watch?v=nctibe_a4rM&list=PLpV0KfVOMojYywq1Blp3SGx62cSyoRfHt&index=3) as a group of designe ...
+The [Bitcoin Design Community]({{ site.slack_invite_url }}) ran a design sprint on an open-source lightning wallet.  During this sprint, we collaboratively focused on re-designing one aspect of the user flow, the **onboarding journey**. In this collaborative effort, [we got together](https://www.youtube.com/watch?v=nctibe_a4rM&list=PLpV0KfVOMojYywq1Blp3SGx62cSyoRfHt&index=3) as a group of designe ...
```
I did not change these lines, not sure why they show as changes.

GBKS replied:
Your code editor may have automatically removed the trailing space at the end of line.

GBKS commented on guide/case-studies/payjoin.md:
```diff
@@ -0,0 +1,311 @@
+
+{% include next-previous.html
+   previousUrl = "/guide/case-studies/blixt-wallet"
+   previousName = "Case studies"
```
```suggestion
   previousName = "Blixt wallet case study"
```

GBKS commented:
@yashrajd thanks for patiently addressing all my nitpicks. I think the case study look great and is ready to go, I'm happy to approve and merge now, you just need to resolve the conflict with the `_compress_images_cache.yml` file. Let me know if you'd like help with that.

yashrajd commented:
> @yashrajd thanks for patiently addressing all my nitpicks. I think the case study look great and is ready to go, I'm happy to approve and merge now, you just need to resolve the conflict with the `_compress_images_cache.yml` file. Let me know if you'd like help with that.

That's great, would love to help resolve the conflict. Looks like my branch is missing the images from the PIN reminders PR?

danielnordh reviewed (approved):
LGTM
Fixed the merge conflict in the cache file

danielnordh reviewed (approved):
LGTM

GBKS reviewed (approved):
ACK.
